app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt

```diff
@@ -30,44 +41,71 @@ import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.BitmapPainter
 import androidx.compose.ui.graphics.vector.ImageVector
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
 import androidx.compose.ui.text.font.FontStyle
 import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.Dp
 import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
 import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.lifecycle.compose.collectAsStateWithLifecycle
 import com.pvp.app.R
-import com.pvp.app.model.User
+import com.pvp.app.model.Ingredient
+import com.pvp.app.model.SportActivity
+import com.pvp.app.ui.common.ProgressIndicator
 import com.pvp.app.ui.common.showToast
 import com.pvp.app.ui.common.underline
 
 @Composable
 private fun ProfileBody(
     modifier: Modifier = Modifier,
-    user: State<User?>,
+    state: State<ProfileState>,
+    context: Context = LocalContext.current,
     onUpdateMass: (Int) -> Unit,
     onUpdateHeight: (Int) -> Unit,
+    onUpdateIngredients: (List<String>) -> Unit,
+    onUpdateActivities: (List<String>) -> Unit,
 ) {
-    val context = LocalContext.current
+    val allActivities = SportActivity
```

Comment:
I believe it would be better if this was a singleton or remembered value to disable recomputation upon rerender.